pkg/selinux/setroubleshoot-client.js

```diff
@@ -35,9 +35,10 @@ var busNameFixit = "org.fedoraproject.SetroubleshootFixit";
 var dbusInterfaceFixit = busNameFixit;
 var dbusPathFixit = "/org/fedoraproject/SetroubleshootFixit/object";
 
-client.init = function() {
+client.init = function(capabilitiesChangedCallback) {
```

Comment:
> Does it? Already we access two different dbus interfaces and we might get more. Worrying about synchronizing that here seems unnecessary, if we can just triggering an update whenever we get more info. The callback implementation seems more robust to me.

We only ever get info once, when we read introspection data. When should that ever change?

> Do you have a better solution? The "old" interface reflects the state of fedora-23, which we support. The newest one with all the features we use is in fedora-24 testing. Since the new features aren't essential, I wouldn't want to make cockpit incompatible with people who don't use the setroubleshoot copr for bleeding edge compatibility.

You're right, this is the most pragmatic solution. Especially when it's only temporary.